metricbeat/module/nats/connections/data.go

```diff
@@ -46,22 +46,17 @@ func eventMapping(r mb.ReporterV2, content []byte) error {
 
 	err := json.Unmarshal(content, &inInterface)
 	if err != nil {
-		err = errors.Wrap(err, "failure parsing Nats connections API response")
-		r.Error(err)
-		return err
+		return errors.Wrap(err, "failure parsing Nats connections API response")
 	}
 	event.MetricSetFields, err = connectionsSchema.Apply(inInterface)
 	if err != nil {
-		err = errors.Wrap(err, "failure applying connections schema")
-		r.Error(err)
-		return err
+		return errors.Wrap(err, "failure applying connections schema")
+
 	}
 
 	event.ModuleFields, err = moduleSchema.Apply(inInterface)
 	if err != nil {
-		err = errors.Wrap(err, "failure applying module schema")
-		r.Error(err)
-		return err
+		return errors.Wrap(err, "failure applying module schema")
 	}
 	r.Event(event)
```

Comment:
Should we add the check here for `report.Event`? If you want to open a separate PR for that, I guess that's fine too 😄

Reply:
I don't think it is needed? this metricset will report a single event on every fetch, so there is no need to check for stop in case of stop